src/legacy/core_plugins/metrics/public/components/annotations_editor.js

```diff
@@ -58,26 +64,46 @@ function newAnnotation() {
 
 const RESTRICT_FIELDS = [ES_TYPES.DATE];
 
+const localStorage = new Storage(window.localStorage);
 class AnnotationsEditor extends Component {
-
   constructor(props) {
     super(props);
+    this.state = {
+      indexPatternForQuery: {},
```

Comment:
I see the `'QueryBar'` works with the index pattern object but not sure that we should add one more field for storing index pattern value.

I offer you 2 different ways: 
1) create a wrapper upper the '**QueryBar**' component and move all this logic related to **uiQueryLanguage** and **indexPatternForQuery**  to this component.  It helps you to remove all code duplication
2) more complex way.  TSVB should work with index pattern object instead of string value of index pattern. But I see a lot of changes here.

@timroes  @markov00  what do you think?